flink-fs-tests/src/test/java/org/apache/flink/hdfstests/ContinuousFileMonitoringTest.java

```diff
@@ -440,10 +491,50 @@ public void testFileSplitMonitoringProcessOnce() throws Exception {
 
 	private int getLineNo(String line) {
 		String[] tkns = line.split("\\s");
-		Assert.assertEquals(tkns.length, 6);
+		Assert.assertEquals(6, tkns.length);
 		return Integer.parseInt(tkns[tkns.length - 1]);
 	}
 
+	private class TimeUpdatingThread extends Thread {
+
+		private volatile boolean isRunning;
+
+		private final TestTimeServiceProvider timeServiceProvider;
+		private final OneInputStreamOperatorTestHarness testHarness;
+		private final long wmInterval;
+		private final int elementUntilUpdating;
+
+		TimeUpdatingThread(final TestTimeServiceProvider timeServiceProvider,
+						   final OneInputStreamOperatorTestHarness testHarness,
+						   final long wmInterval,
+						   final int elementUntilUpdating) {
+
+			this.timeServiceProvider = timeServiceProvider;
+			this.testHarness = testHarness;
+			this.wmInterval = wmInterval;
+			this.elementUntilUpdating = elementUntilUpdating;
+			this.isRunning = true;
+		}
+
+		@Override
+		public void run() {
+			try {
+				while (isRunning) {
+					if (testHarness.getOutput().size() % elementUntilUpdating == 0) {
```

Comment:
There is a "race" between the operator emitting elements and this thread. Both run in loops without delays. Only if this condition is evaluated by chance at the exact point in time when the list happens to have so many result elements, there will actually be a time advance.